test/metabase/driver/postgres_test.clj

```diff
@@ -1303,3 +1320,22 @@
     ;; Special characters
     (is (= "SET ROLE \"Role.123\";"   (driver.sql/set-role-statement :postgres "Role.123")))
     (is (= "SET ROLE \"$role\";"      (driver.sql/set-role-statement :postgres "$role")))))
+
+(deftest sync-row-count-test
+  (mt/test-driver :postgres
+    (testing "Can sync row count"
+      (mt/with-temp-test-data ["city"
+                               [{:field-name "name"
+                                 :base-type  :type/Text}]
+                               [["Los Angeles"]
+                                ["Las Vegas"]]]
+        (sql-jdbc.execute/do-with-connection-with-options
+         :postgres
+         (mt/db)
+         nil
+         (fn [conn]
+           ;; row count are estimated so we VACUUM so the statistic table is updated before syncing
```

Comment:
I'm not sure how precise you expect the stats to be afterward a vacuum, but as far as I know it won't directly update them and the improvements happening will be indirect through updating things like the visibility map. I think you would want to call `ANALYZE`. Even then I believe it does sampling and there are no guarantees at scale.